ppr-ui/src/components/unitNotes/UnitNoteReview.vue

```diff
@@ -1,23 +1,322 @@
+      isRoleStaffReg
+    } = storeToRefs(useStore())
+
+    const {
+      setMhrUnitNote,
+      setMhrUnitNoteRegistration,
+      setStaffPayment
+    } = useStore()
+
+    const {
+      getValidation,
+      setValidation
+    } = useMhrValidations(toRefs(getMhrUnitNoteValidation.value))
+
     const localState = reactive({
+      validate: false,
+      validateSubmittingParty: false,
+      unitNoteType: UnitNotesInfo[getMhrUnitNote.value.documentType],
+      givingNoticeParty: computed((): PartyIF => getMhrUnitNote.value.givingNoticeParty),
+      unitNoteSubmittingParty: computed(() => getMhrUnitNoteRegistration.value.submittingParty || {}),
+      staffPayment: {
+        option: StaffPaymentOptions.NONE,
+        routingSlipNumber: '',
+        bcolAccountNumber: '',
+        datNumber: '',
+        folioNumber: '',
+        isPriority: false
+      },
+      validateStaffPayment: computed(() => {
+        return isRoleStaffReg.value &&
+        localState.validate &&
+        !getValidation(MhrSectVal.UNIT_NOTE_VALID, MhrCompVal.STAFF_PAYMENT_VALID)
+      })
     })
 
+    const setGivingNoticeParty = (val: PartyIF) => {
+      setMhrUnitNote({ key: 'givingNoticeParty', value: val })
+    }
+
+    const setSubmittingParty = (val: SubmittingPartyIF) => {
+      setMhrUnitNoteRegistration({ key: 'submittingParty', value: val })
+    }
+
+    const handleComponentValid = (component: MhrCompVal, isValid: boolean) => {
+      setValidation(MhrSectVal.UNIT_NOTE_VALID, component, isValid)
+    }
+
+    const onStaffPaymentDataUpdate = (val: StaffPaymentIF) => {
+      let staffPaymentData: StaffPaymentIF = {
+        ...val
+      }
+
+      switch (staffPaymentData.option) {
+        case StaffPaymentOptions.FAS:
+          staffPaymentData = {
+            option: StaffPaymentOptions.FAS,
+            routingSlipNumber: staffPaymentData.routingSlipNumber,
+            isPriority: staffPaymentData.isPriority,
+            bcolAccountNumber: '',
+            datNumber: '',
+            folioNumber: ''
+          }
+          break
+
+        case StaffPaymentOptions.BCOL:
+          staffPaymentData = {
+            option: StaffPaymentOptions.BCOL,
+            bcolAccountNumber: staffPaymentData.bcolAccountNumber,
+            datNumber: staffPaymentData.datNumber,
+            folioNumber: staffPaymentData.folioNumber,
+            isPriority: staffPaymentData.isPriority,
+            routingSlipNumber: ''
+          }
+          break
+
+        case StaffPaymentOptions.NO_FEE:
+          staffPaymentData = {
+            option: StaffPaymentOptions.NO_FEE,
+            routingSlipNumber: '',
+            isPriority: false,
+            bcolAccountNumber: '',
+            datNumber: '',
+            folioNumber: ''
+          }
+          break
+        case StaffPaymentOptions.NONE: // should never happen
+          break
+      }
+
+      localState.staffPayment = staffPaymentData
+      setStaffPayment(staffPaymentData)
+    }
```

Comment:
Its kind of a bummer this code is required in all instances of Staff Payment. 
After we upgrade, we should create a shim to consolidate all of this work, so all we need is to instantiate the component.

Reply:
Yea I know... I didn't want to bring it in here either - it really should be part of Staff Payment. Maybe a have our own wrapper?

Reply:
Definitely, not part of your scope here just thinking its a future-us problem to tackle this. 
Probably post upgrade, we can consolidate these various instances of StaffPayment

Reply:
Yea we should clean up after the upgrade. I will create a ticket.